### DISCUSSION TITLE
Adding constraints to state derivative vector

### DISCUSSION THREAD

**Discussion by agrace22** [casadi/casadi#3325] [Q&A]
How would I add constraints on the xdot vector in the direct_multiple_shooting.m example? For example, what would I do if I didn't want xdot(1) to ever go below zero? Would I add the constraints within the section that formulates the NLP?

for k=0:N-1
    % New NLP variable for the control
    Uk = MX.sym(['U_' num2str(k)]); 
    w = {w{:}, Uk};
    lbw = [lbw; -1]; 
    ubw = [ubw;  1]; 
    w0 = [w0;  0];

    % Integrate till the end of the interval
    Fk = F('x0', Xk, 'p', Uk);
    Xk_end = Fk.xf;
    J=J+Fk.qf;

    % New NLP variable for state at end of interval
    Xk = MX.sym(['X_' num2str(k+1)], 2); 
    w = [w, {Xk}];
    lbw = [lbw; -0.25; -inf]; 
    ubw = [ubw;  inf;  inf]; 
    w0 = [w0; 0; 0];

    % Add equality constraint
    g = [g, {Xk_end-Xk}];
    lbg = [lbg; 0; 0];
    ubg = [ubg; 0; 0];
end

**Comment by brunoml5** [comment]
Hi!

If you want to implement path constraints, just write your expression and put it in `g`, don't mess with the equality constraint of the multiple-shooting method. 

Note that for the multiple-shooting method, the path constraint will only be enabled at the beginning/end of each interval. Maybe switch to collocation because you will be able to implement this constraint for all integration steps.

I hope it helps!

**Reply by agrace22** [reply]
Would this be satisfactory for putting a lower limit of 0 on the entry of xdot(1)?

% Add equality constraint
g = [g, {Xk_end-Xk}, {xdot(1)}];
lbg = [lbg; 0; 0; 0 ];
ubg = [ubg; 0; 0; inf];

**Reply by brunoml5** [reply]
Yeah, it should be enough. Try it. But also consider switching to the collocation method.
